specifications/tcp/tcp.tla

```diff
@@ -0,0 +1,330 @@
+------------------------------- MODULE tcp ----------------------------------
+(* 
+https://datatracker.ietf.org/doc/html/rfc9293
+
+This spec abstracts from many details such as sequence numbers in RFC9293 and
+focuses on the state transitions of the TCP finite state machine shown in the
+diagram at the end of this file.
+*)
+EXTENDS Integers, Sequences, SequencesExt, FiniteSets
+
+CONSTANT 
+    Peers
+
+ASSUME Cardinality(Peers) = 2
+
+States == {"LISTEN", "CLOSED", "SYN-SENT", "SYN-RECEIVED", "ESTABLISHED", 
+          "FIN-WAIT-1", "FIN-WAIT-2", "CLOSING", "CLOSE-WAIT", "LAST-ACK",
+          "TIME-WAIT"}
+
+VARIABLE
+    tcb,
+    connstate,
+    network
+
+vars == <<tcb, connstate, network>>
+
+TypeOK ==
+    /\ tcb \in [ Peers -> BOOLEAN ]
+    /\ connstate \in [ Peers -> States ]
+    /\ network \in [ Peers -> Seq({"SYN", "SYN,ACK", "ACK", "RST", "FIN"} \cup {"ACKofFIN"}) ]
+
+Init ==
+    /\ tcb = [p \in Peers |-> FALSE]
+    /\ connstate = [p \in Peers |-> "CLOSED"]
```

Comment:
With this initial condition, does your action constraint still make sense, given that ACTIVE_OPEN and PASSIVE_OPEN are the only actions enabled from connection state CLOSED?

Reply:
With TLC, `Init` is redefined by `MCInit` in `MCtcp.tla`.